unit/memleak_hawaii3.py

```diff
@@ -39,13 +39,23 @@
     ax.pcolor(10 * rand(50, 50))
 
     fig.savefig('tmp%d' % i, dpi=75)
-    close(1)
+    plt.close(1)
 
-    gc.collect()
+    coll = gc.collect()
     val = report_memory(i)
     print(i, val)
     if i == indStart:
         start = val  # wait a few cycles for memory usage to stabilize
+    mem_size.append(val)
+    coll_count.append(coll)
 
 end = val
-print('Average memory consumed per loop: %1.4fk bytes\n' % ((end - start) / float(indEnd - indStart)))
+print('Average memory consumed per loop: %1.4fk bytes\n' %
+      ((end - start) / float(indEnd - indStart)))
+fig, ax = plt.subplots()
+ax2 = ax.twinx()
+ax.plot(mem_size, 'r')
+ax.set_ylabel('memory size', color='r')
+ax2.plot(coll_count, 'k')
+ax2.set_ylabel('collect count', color='k')
+fig.savefig('report')
```

Comment:
Somehow, we need to make checking for memory leakage standard, not necessarily on every commit, but at least something that can be triggered weekly.  In my opinion, memory leaks are such fundamental errors that they should have top priority for attention.  Easy for me to say, though; tracking them down is not something for which I have much competence or time.